### PR TITLE
Austenem/CAT-785 add datasets section

### PR DESCRIPTION
**Pull request by austenem**
## Summary

This update adds a datasets section with an info banner to empty workspace detail pages, allowing users to add datasets within the page or via the search page. Minor style changes were applied to the workspaces detail page in order to match intended designs. 

## Design Documentation/Original Tickets

[CAT-785 JIRA ticket](https://hms-dbmi.atlassian.net/browse/CAT-785?atlOrigin=eyJpIjoiZDc1OGRmMmY0Yjg1NDVlMjlkYjg5YWRlNTc5ZTIwYjUiLCJwIjoiaiJ9)

[Figma mockup](https://www.figma.com/design/NgBjvs0jRbDHhiyn5nicDm/Workspace?node-id=2691-1160&m=dev)

## Testing

Feature was manually tested by ensuring that the new info banner only appears on workspace detail pages without datasets, and that the two new buttons redirect the user appropriately to the datasets search page and "Add Datasets to Workspace" modal. 

## Screenshots/Video

Screenshot 2024-07-29 at 10 55 41 AM

## Checklist

- [X] Code follows the project's coding standards
  - [X] Lint checks pass locally
  - [X] New `CHANGELOG-your-feature-name-here.md` is present in the root directory, describing the change(s) in full sentences.
- [ ] Unit tests covering the new feature have been added
- [X] All existing tests pass
- [X] Any relevant documentation in JIRA/Confluence has been updated to reflect the new feature
- [X] Any new functionalities have appropriate analytics functionalities added

### CODE REVIEW

File: context/app/static/js/pages/Workspace/Workspace.tsx [hunk 1]
```diff
@@ -47,6 +52,52 @@ function WorkspaceContent({ workspaceId }: WorkspacePageProps) {
...
+          }
+        />
+        <Alert
+          severity="info"
+          action={
+            <Button>
+              <InternalLink href="/search?entity_type[0]=Dataset">
+                <Typography color="primary" variant="button">
+                  {' '}
+                  Dataset Search Page{' '}
```
Comment (john-conroy) [inline context/app/static/js/pages/Workspace/Workspace.tsx:90]: Do we need these empty spaces?

File: context/app/static/js/pages/Workspace/Workspace.tsx [hunk 2]
```diff
@@ -47,6 +52,52 @@ function WorkspaceContent({ workspaceId }: WorkspacePageProps) {
 
   const job = condenseJobs(workspace.jobs);
 
+  const datasetsSection =
+    workspaceDatasets.length > 0 ? (
+      <WorkspaceDatasetsTable
+        datasetsUUIDs={workspaceDatasets}
+        label={<SectionHeader> Datasets</SectionHeader>}
```
Comment (john-conroy) [inline context/app/static/js/pages/Workspace/Workspace.tsx:59]:
```suggestion
        label={<SectionHeader>Datasets</SectionHeader>}
```

File: context/app/static/js/pages/Workspace/Workspace.tsx [hunk 3]
```diff
@@ -47,6 +52,52 @@ function WorkspaceContent({ workspaceId }: WorkspacePageProps) {
...
+          <WorkspacesUpdateButton
+            workspace={workspace}
+            dialogType="ADD_DATASETS"
+            sx={(theme) => ({
+              marginRight: theme.spacing(1),
+            })}
+            tooltip={tooltips.datasets}
+          >
+            <AddIcon />
+          </WorkspacesUpdateButton>
```
Comment (john-conroy) [inline context/app/static/js/pages/Workspace/Workspace.tsx:70]: Can we dedupe some of the shared components/logic between these two cases?

Comment (john-conroy) [reply context/app/static/js/pages/Workspace/Workspace.tsx:70]: We could just add a `boolean` prop to `WorkspaceDatasetsTable` to choose whether or not to show the copy button similar to how we handle `removeDatasets`. We could then display either the alert or the table based on whether the `datasetsUUIDs` passed to `WorkspaceDatasetsTable` is empty.

File: context/app/static/js/pages/Workspace/Workspace.tsx [hunk 4]
```diff
@@ -47,6 +52,52 @@ function WorkspaceContent({ workspaceId }: WorkspacePageProps) {
 
   const job = condenseJobs(workspace.jobs);
 
+  const datasetsSection =
```
Comment (john-conroy) [inline context/app/static/js/pages/Workspace/Workspace.tsx:55]: We should pull this logic into `WorkspaceDatasetsTable` or another component.

**Review by john-conroy — CHANGES_REQUESTED** [review]
Thanks! A few comments. If at any point it would be helpful to talk through some of this let me know!

File: context/app/static/js/pages/Workspace/Workspace.tsx (hunk 1 above)
Comment (austenem) [reply context/app/static/js/pages/Workspace/Workspace.tsx:90]: Nope, think these were added by eslint

File: context/app/static/js/pages/Workspace/Workspace.tsx (hunk 3 above)
Comment (austenem) [reply context/app/static/js/pages/Workspace/Workspace.tsx:70]: I think this all makes sense, but I'd like to go over how I moved this logic into `WorkspaceDatasetsTable` - it alters how the empty table displays in other usages as well.

File: context/app/static/js/components/workspaces/WorkspaceDatasetsTable/WorkspaceDatasetsTable.tsx [hunk 5]
```diff
@@ -45,28 +61,46 @@ function WorkspaceDatasetsTable({
     }),
     [datasetsUUIDs],
   );
+
+  const datasetsPresent = datasetsUUIDs.length > 0;
```
Comment (john-conroy) [inline context/app/static/js/components/workspaces/WorkspaceDatasetsTable/WorkspaceDatasetsTable.tsx:65]: Prefer this to be `hasDatasets`, but not blocking!

**Review by john-conroy — APPROVED** [review]
Thanks for the updates. Looks good!

**Pull request merged**